command/agent/job_endpoint.go

```diff
@@ -1319,21 +1319,7 @@ func ApiTaskToStructsTask(job *structs.Job, group *structs.TaskGroup,
 		}
 	}
 
-	if len(apiTask.VolumeMounts) > 0 {
-		structsTask.VolumeMounts = []*structs.VolumeMount{}
-		for _, mount := range apiTask.VolumeMounts {
-			if mount != nil && mount.Volume != nil {
-				structsTask.VolumeMounts = append(structsTask.VolumeMounts,
-					&structs.VolumeMount{
-						Volume:          *mount.Volume,
-						Destination:     *mount.Destination,
-						ReadOnly:        *mount.ReadOnly,
-						PropagationMode: *mount.PropagationMode,
-						SELinuxLabel:    *mount.SELinuxLabel,
-					})
-			}
-		}
-	}
+	structsTask.VolumeMounts = apiVolumeMountsToStructs(apiTask.VolumeMounts)
```

Comment:
nice!

Reply:
Will do.

Reply:
Good thing you pointed this out -- I should have been using `EqualFunc` in `nomad/structs/services.go`!